test/testdata/infer/exhaustiveness.rb

```diff
@@ -164,6 +160,74 @@ def cant_call_only_absurd(x)
   T.absurd(x) # error: This code is unreachable
 end
 
+# --- reasonable usage on global, class, instance, and local variables --------
+$some_global = T.let(true, T::Boolean)
+
+sig {void}
+def absurd_not_reached_on_global_var
+  if $some_global
+    return
+  else
+    return
+  end
+
+  T.absurd($some_global)
```

Comment:
wow i think this is the first usage of `T.absurd` like this, completely outside of an `if / else` 😅 
nice to see it just works

Reply:
#TrustTheCFG